modules/wsdl4j/src/main/java/com/ibm/wsdl/extensions/PopulatedExtensionRegistry.java

```diff
@@ -23,428 +24,425 @@
  *
  * @author Matthew J. Duftler (duftler@us.ibm.com)
  */
-public class PopulatedExtensionRegistry extends ExtensionRegistry
-{
-  public static final long serialVersionUID = 1;
-
-  public PopulatedExtensionRegistry()
-  {
-    init();
-    handleSchemaContent();
-  }
-
-  public PopulatedExtensionRegistry(boolean ignoreSchemaContent)
-  {
-    init();
-    if(!ignoreSchemaContent) {
-      handleSchemaContent();
+public class PopulatedExtensionRegistry extends ExtensionRegistry {
+    public static final long serialVersionUID = 1;
+
+    public PopulatedExtensionRegistry() {
+        init();
```

Comment:
Can we call the overloaded constructor here

Reply:
I don't think it is the correct way of doing it. we usually call the super constructor in the overloaded constructors. This is to call it vice versa. in this case, it would bring no additional value to the code as both constructors do need to be public.

Reply:
"The benefit of having one constructor invoke another overloaded constructor is to avoid code duplication."
" Basically, each of the other not-the-real-one overloaded constructors will call another overloaded constructor, passing it whatever data it needs (data the client code didn't supply)."

pp 144, SCJP Sun® Certified Programmer for Java™ 6 Study Guide (Exam 310-065)

The practice is to implement the logic at a single place (where possible, to avoid code duplication), in this case, public PopulatedExtensionRegistry(boolean ignoreSchemaContent), and invoke it from the public PopulatedExtensionRegistry().
public PopulatedExtensionRegistry() {
    this(false);
}

Reply:
You are right, there's a concept called Constructor chaining, It is also referred to as the Telescoping Constructor Pattern, and is now considered to be an Anti Pattern. 

Please refer to pp 11 of  Effective Java™, Second Edition. Or there are plenty of google references on the Anti pattern.

To handle this there are other alternatives. But I believe, at this stage, handling it otherwise is not really necessary or a must do fix.

Reply:
Constructor Chaining and Telescoping Constructors are two different things. Yes, Telescoping Constructors is an anti-pattern. But Constructor Chaining is an inbuilt feature in the language. You can't even compile a simplest Java class without constructor chaining.

The suggestion was given to reuse the existing code and avoid duplicate code.

Reply:
Constructor chaining, calling the default constructor is correct and also by using super and this to call the default constructor is also Constructor chaining, I believe you are referring to that when you say that a simplest java class cannot compile without constructor chaning. But the  way (The method you are referring to is also constructor chaining AFAIU) you are proposing is to exactly how the Telescoping Constructor pattern is described. Could you please let me know what the difference is between what you are proposing and what Telescoping constructor pattern describes? Probably I am not getting your point here.

Reply:
Telescoping is calling many Constructors in sequence C()->C(arg)->C(arg1, arg2)->C(arg1, arg2, arg3) or the reverse of this... which might mislead the user and which is sometimes error prone. A user might sometimes mistakenly miss one of the arguments, but it matches with another Constructor available. That is one reason to consider Telescoping Constructor an anti-pattern. However, this was done by earlier developers to avoid code duplication (which is good), But the proper solution for this is to follow builder pattern or something similar when you have many constructors. 

However, in the above code, it could have avoided duplicate codes just calling the overloaded constructor. And it doesn't introduce any ambiguity to the code. 

Anyway, as we have a limited number of duplicate lines in both the constructors. I'll merge this code for the moment. But in the future, if a requirement arises to add more logic to the constructor we have to reconsider this and refactor accordingly. Because duplicate code might lead to a lot of issues.

Reply:
Thank you for merging the PR. However, I still feel what you are asking me to do, to avoid code duplication is also Telescoping. That is calling C() ->C(arg).  The reasons you have mentioned on why it is a Anti pattern is understood, and it is for the same reason you have mentioned, which is code duplication. But your explanation on it is understood and is the same as my understanding. Thank you for taking the time to explain.

I believe, you should make code simpler than complex by choosing the appropriate design/implementation concepts that suits the requirement, rather than following it in the name of standard. Hence the reluctance, and not being convinced.

Yes,  agreed, in future, if we do have any other requirements that do arise to add more logic, we can always re-factor the code. Lets do it then, rather than over-engineer this now.